mindsdb_sql/parser/dialects/mindsdb/knowledge_base.py

```diff
@@ -0,0 +1,78 @@
+from mindsdb_sql.parser.ast.base import ASTNode
+from mindsdb_sql.parser.utils import indent
+
+
+class CreateKnowledgeBase(ASTNode):
+    def __init__(
+        self,
+        name,
+        model,
+        storage,
+        from_select=None,
+        params=None,
+        if_not_exists=False,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.name = name
+        self.model = model
+        self.storage = storage
+        self.params = params
+        self.if_not_exists = if_not_exists
+        self.from_query = from_select
+
+    def to_tree(self, *args, level=0, **kwargs):
+        ind = indent(level)
+        out_str = f"""
+        {ind}CreateKnowledgeBase(
+        {ind}    if_not_exists={self.if_not_exists},
+        {ind}    name={self.name.to_string()},
+        {ind}    from_query={self.from_query.to_tree(level=level+1) if self.from_query else None},
+        {ind}    model={self.model.to_string()},
+        {ind}    storage={self.storage.to_string()},
+        {ind}    params={self.params}
+        {ind})
+        """
+        return out_str
+
+    def get_string(self, *args, **kwargs):
+        params = self.params.copy()
+        using_ar = [f"{k}={repr(v)}" for k, v in params.items()]
+        using_str = ", ".join(using_ar)
+        from_query_str = (
+            f"FROM ({self.from_query.get_string()})" if self.from_query else ""
+        )
+
+        out_str = (
+            f"CREATE KNOWLEDGE_BASE {'IF NOT EXISTS' if self.if_not_exists else ''}{self.name.to_string()} "
```

Comment:
The syntax is coming from: https://docs.google.com/document/d/1_-Gc5yIeuXDeXPeBqhyY5y-yk6FJmNyCe1Qr7jtCVh0/edit#heading=h.hatc4xi8koc2

```
CREATE KNOWLEDGE BASE <knowledge_base_name>
FROM 
(
	SELECT id, review as content
	FROM demo_db.amazon_reviews
)
USING
	STORAGE = <vector_database_name>.<collection_name>, -- these are references
	MODEL = <embedding_model_name>
; 
```

The high-level goal is to use the identifier of the model and the vector database table directly (instead of strings). 
I do not have a strong preference on the syntax either. 

I think how `USING` clause is parsed now does not allow that we pass identifiers directly. So we need to use strings to refer to names.
But I do see in the implementation of the chatbot that these strings get translated into identifiers when actually creating the tokens.

Reply:
Made the syntax change. 
Users will be able use either of the two
```
CREATE KNOWLEDGE BASE my_kb
USING
  MODEL = <embeddng_model_name>,
  STORAGE = <vector_database_name>.<collection_name>

```
or
```
CREATE KNOWLEDGE BASE my_kb
USING
  MODEL = '<model_name>'
  STORAGE = '<vector_database_name>.<collection_name>'
```